hilo.rb

```diff
@@ -1,34 +1,121 @@
 #!/usr/bin/env ruby
-puts "Welcome to the Hi-Lo game.\nGuess a random whole number from 1 to 100 (inclusive)."
-puts "You will have a total of 6 guesses."
-puts ""
 
-random_num = rand(1..100)
+# Game to guess random number
+class HiLoGame
 
-# gives the user 6 guesses
-for i in 0..5
-    guesses_left = 6 - i
-    puts "You have #{guesses_left} guesses left."
-    puts "Please guess a whole number."
+  # Constructor to initialise the class with a maximum number of guesses
+  def initialize(max_guesses)
+    @max_guesses = max_guesses
+    @already_guessed = Array.new
+    @random_num = get_random_num
+  end
 
-    # only allow integer input
+  # The game is started
+  def play
+    welcome
+
+    begin
+      print_guesses_left
+      guess = guess_a_number
+      check_guess(guess)
+    end until !game_running?(guess)
+
+    finish_game(guess)
```

Comment:
You shouldn't be accessing the `guess` variable outside the loop, I know you can but it makes it a bit confusing tracing back through code.

Anyway this is getting better but it's still not as clear as it could be. Try getting something like this working
```
while game_running? #with no guess passed in
  ...
end

finish_game # also with no guess
```

Reply:
Hint: you don't need to know the previous guess because you know all the guesses.

Reply:
Done in https://github.com/amykingOW/Hi-Lo-Game/pull/4/commits/6451208e3d490a7109a160b8396732d668ab14fb.